File: src/utils/load-api-config.mjs
// Shared utility: parse an api.env file and return DataForSEO credentials.
// Used by fetch-serp.mjs and fetch-keywords.mjs.

import { readFileSync } from 'node:fs';

/**
 * Parse an api.env file into { auth, base }.
 * Skips empty lines and comment lines (starting with #).
 * Splits each line on the first `=` only, so values may contain `=`.
 * @param {string} filePath - absolute path to the env file
 * @returns {{ auth: string, base: string }}
 */
function loadEnv(filePath) {
  const content = readFileSync(filePath, 'utf-8');
  const env = {};
  for (const line of content.split('\n')) {
    const trimmed = line.trim();
    if (trimmed === '' || trimmed.startsWith('#')) continue;
    const eqIdx = trimmed.indexOf('=');
    if (eqIdx === -1) continue;
    env[trimmed.slice(0, eqIdx)] = trimmed.slice(eqIdx + 1);
  }

  const auth = env.DATAFORSEO_AUTH;
  const base = env.DATAFORSEO_BASE;
  if (auth === undefined || auth === '') {
    throw new Error('DATAFORSEO_AUTH must be set in api.env');
  }
  if (base === undefined || base === '') {
    throw new Error('DATAFORSEO_BASE must be set in api.env');
  }
  return { auth, base };
}

export { loadEnv };

File: seo_pipeline/utils/load_api_config.py
"""Parse an api.env file and return DataForSEO credentials."""

from pathlib import Path

from dotenv import dotenv_values


def load_env(file_path: str) -> dict:
    """
    Parse an api.env file into { 'auth': str, 'base': str }.

    Args:
        file_path: absolute path to the env file

    Returns:
        A dict with keys 'auth' and 'base'

    Raises:
        ValueError: if DATAFORSEO_AUTH or DATAFORSEO_BASE are missing or empty
        FileNotFoundError: if the file does not exist
    """
    path = Path(file_path)
    if not path.exists():
        raise FileNotFoundError(f"No such file or directory: '{file_path}'")

    env = dotenv_values(file_path)

    auth = env.get('DATAFORSEO_AUTH')
    base = env.get('DATAFORSEO_BASE')

    if not auth:
        raise ValueError('DATAFORSEO_AUTH must be set in api.env')
    if not base:
        raise ValueError('DATAFORSEO_BASE must be set in api.env')

    return {'auth': auth, 'base': base}
